src/backend/commands/tablecmds.c

```diff
@@ -17666,6 +17666,12 @@ static Datum transformExecOnClause(List	*on_clause, int *preferred_segment_num)
 		}
 		else if (strcmp(defel->defname, "master") == 0)
 		{
+			if(iswritable){
+			  ereport(ERROR,
+	 	      (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+			    errmsg("the ON master syntax for writable external tables is deprecated"),
```

Comment:
Only ON number is allowed, e.g. ON 3 will start 3 virtual segments to run the query.

Here's an example to create a writable external table in Hawq 2.0:
postgres=# create table a (i int);
CREATE TABLE
postgres=# insert into a values(1);
INSERT 0 1
postgres=# insert into a values(2);
INSERT 0 1
postgres=# insert into a values(10);
INSERT 0 1
postgres=# CREATE WRITABLE EXTERNAL WEB TABLE externala (output text)         EXECUTE 'cat > /tmp/externala' on 3 FORMAT 'TEXT'   DISTRIBUTED RANDOMLY;
CREATE EXTERNAL TABLE
postgres=# insert into externala select * from a;
INSERT 0 3

You can check the result file in /tmp/externala:
1
2
10